lib/galaxy/tool_shed/galaxy_install/tools/data_manager.py

```diff
@@ -60,16 +59,15 @@ def install_data_managers(self, shed_data_manager_conf_filename, metadata_dict,
             repository_tools_by_guid = {}
             for tool_tup in repository_tools_tups:
                 repository_tools_by_guid[tool_tup[1]] = dict(tool_config_filename=tool_tup[0], tool=tool_tup[2])
+
+            # If conf file does not exist, create it.
+            if not os.path.exists(shed_data_manager_conf_filename):
+                with open(shed_data_manager_conf_filename, 'w') as fh:
+                    log.info('Creating shed data manager config with default contents: %s', shed_data_manager_conf_filename)
+                    fh.write(SHED_DATA_MANAGER_CONF_XML)
+
             # Load existing data managers.
-            try:
-                tree, error_message = parse_xml(shed_data_manager_conf_filename, check_exists=False)
-            except (OSError, IOError) as exc:
-                if exc.errno == errno.ENOENT:
-                    with open(shed_data_manager_conf_filename, 'w') as fh:
-                        fh.write(SHED_DATA_MANAGER_CONF_XML)
-                    tree, error_message = parse_xml(shed_data_manager_conf_filename)
-                else:
-                    raise
+            tree, error_message = parse_xml(shed_data_manager_conf_filename)
```

Comment:
My suggestion (for both this file and `lib/galaxy/tool_shed/galaxy_install/tools/tool_panel_manager.py`) would be to replace the use of `parse_xml()` from `galaxy.util.tool_shed.xml_util` to the one from `galaxy.util` (with `remove_comments=False, strip_whitespace=False` as needed).

Reasons:
- `error_message` is not checked and basically unused
- it would be better to let the exception bubble up if there is a syntax error in the XML file.

Reply:
OK, trying this.